### PR TITLE
feat(sui-hoc): intersection-observer hoc

### PR DESCRIPTION
**Pull request by juanmaguitar**

### CODE REVIEW

File: packages/sui-hoc/package.json [hunk 1]
```diff
@@ -14,6 +14,9 @@
     "prop-types": "15",
     "react": "15 || 16"
   },
+  "dependencies": {
+    "intersection-observer": "0"
```
Comment (jcger) [inline packages/sui-hoc/package.json]: please use the exact version

File: packages/sui-hoc/src/withIntersectionObserver.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/* global IntersectionObserver */
+import React, {Component} from 'react'
+
+if (typeof window !== 'undefined' && window.document) {
```
Comment (midudev) [inline packages/sui-hoc/src/withIntersectionObserver.js]: What about using a dynamic import only if the browser doesn't have support for it. So the check would be done in the componentDidMount, which is only executed in the client.

Comment (juanmaguitar) [reply packages/sui-hoc/src/withIntersectionObserver.js]: With this code 
```
if (typeof window !== 'undefined' && window.document) {
  require('intersection-observer')
}
```
we assure we only load the library/pollyfill in the client. 

For what I've seen in other implementations in Schibsted, the general criteria is to load always the pollyfill and let that code to check if the specific feature is available and apply proper changes if not

What exactly is your suggestion here?

**Comment by juanmaguitar** [comment]
@midudev I have added the code to export also the version that will allow the developer to customize the intersection observer w/ custom options

File: packages/sui-hoc/src/withIntersectionObserver.js (hunk 2 above)
Comment (carlosvillu) [reply packages/sui-hoc/src/withIntersectionObserver.js]: using that IF you will end with the polyfill in your bundler. You have to use the `import()` statement to have lazy loading of this polyfill.

**Pull request merged**